### PR TITLE
[SQL] Always generate code using handles for circuit inputs/outputs

### LINKED ISSUE #1204: [SQL] Convert the SQL compiler tests to use the handle Rust API

### LINKED ISSUE #1207: There is no Rust de/serialization implemented for several SQL data types

### PR DESCRIPTION
**Pull request by mihaibudiu**
Is this a user-visible change (yes/no): no

Fixes #1207 
Fixes #1204 

Previously the Rust generated by the compiler had two different structures depending on whether we test or deploy as a pipeline. With this PR we bring the two structures much closer. There is still a difference: tests use a strongly-typed API for the stream handles that are the circuit I/Os, while the pipeline uses the Catalog API to obtain untyped stream handles.

Merging this properly would require a fix for #1208. In the meantime I have disabled the check that the panic message is adequate.

### CODE REVIEW

**Comment by mihaibudiu** [comment]
@ryzhyk this has some Rust code, please review. Some of the changes may clash with the fixes planned for #1208.

**Comment by mihaibudiu** [comment]
@lalithsuresh I will spare you a review request for this one, the changed code isn't really readable since it essentially generates Rust strings. The fact that the tests pass is a good reassurance that it does the right thing. I will self-review the Java parts myself.

File: crates/dbsp/src/circuit/runtime.rs [hunk 1]
```diff
@@ -26,6 +26,7 @@ pub enum Error {
     /// Panic in a worker thread.
     WorkerPanic {
         worker: usize,
+        message: String,
```
Comment (ryzhyk) [inline crates/dbsp/src/circuit/runtime.rs:29]: These changes to `WorkerPanic` will be superseded by my upcoming PR that will add panic information to `WorkerPanic`.

File: sql-to-dbsp-compiler/lib/sqllib/src/interval.rs [hunk 2]
```diff
@@ -66,6 +67,53 @@ where
     }
 }
 
+impl SerializeWithContext<SqlSerdeConfig> for ShortInterval {
```
Comment (ryzhyk) [inline sql-to-dbsp-compiler/lib/sqllib/src/interval.rs:70]: A simpler way to implement this is to annotate the types with [`#serde[transparent]` ](https://serde.rs/container-attrs.html#transparent) and auto-devive `Serialize/Deserialize` impls and then use the `(de)serialize_without_context` macro to derive `(De)SerializeWithContext`

File: sql-to-dbsp-compiler/lib/sqllib/src/lib.rs [hunk 3]
```diff
@@ -1089,3 +1091,45 @@ where
     }
     builder.done()
 }
+
+pub fn append_to_upsert_handle<K, W>(data: &OrdZSet<K, W>, handle: UpsertHandle<K, bool>)
```
Comment (ryzhyk) [inline sql-to-dbsp-compiler/lib/sqllib/src/lib.rs:1095]: This should take `UpsertHandle` by reference.

File: sql-to-dbsp-compiler/lib/sqllib/src/lib.rs [hunk 4]
```diff
@@ -1089,3 +1091,45 @@ where
...
+        while !w.le0() {
+            let key = cursor.key();
+            handle.push(key.clone(), insert);
+            w = w.add(W::neg(W::one()));
+        }
+        cursor.step_key();
+    }
+}
+
+pub fn append_to_collection_handle<K, W>(data: &OrdZSet<K, W>, handle: CollectionHandle<K, W>)
```
Comment (ryzhyk) [inline sql-to-dbsp-compiler/lib/sqllib/src/lib.rs:1117]: Same here

File: sql-to-dbsp-compiler/lib/sqllib/src/lib.rs [hunk 5]
```diff
@@ -1089,3 +1091,45 @@ where
...
+    W: DBWeight + ZRingValue,
+{
+    let mut cursor = data.cursor();
+    while cursor.key_valid() {
+        handle.push(cursor.key().clone(), cursor.weight());
+        cursor.step_key();
+    }
+}
+
+pub fn read_output_handle<K, W>(handle: OutputHandle<OrdZSet<K, W>>) -> OrdZSet<K, W>
```
Comment (ryzhyk) [inline sql-to-dbsp-compiler/lib/sqllib/src/lib.rs:1129]: Same here

File: sql-to-dbsp-compiler/lib/sqllib/src/lib.rs [hunk 6]
```diff
@@ -1089,3 +1091,45 @@ where
...
+{
+    let mut cursor = data.cursor();
+    while cursor.key_valid() {
+        let mut w = cursor.weight();
+        let mut insert = true;
+        if !w.ge0() {
+            insert = false;
+            w = w.neg();
+        }
+        while !w.le0() {
```
Comment (ryzhyk) [inline sql-to-dbsp-compiler/lib/sqllib/src/lib.rs:1108]: This is not incorrect, but it should be sufficient to insert once.

**Review by ryzhyk — CHANGES_REQUESTED** [review]

**Review by ryzhyk — APPROVED** [review]

**Comment by ryzhyk** [comment]
@mihaibudiu , I merged #1215 .  Can you please rebase and test before merging?

**Comment by mihaibudiu** [comment]
Rebased, if the tests pass we should merge.

**Comment by mihaibudiu** [comment]
This CI fails because it looks like the panic errors are not really deterministic
I will file an issue about it.

**Pull request merged**